azure-pipelines.yml

```diff
@@ -0,0 +1,191 @@
+# Starter pipeline
+# Start with a minimal pipeline that you can customize to build and deploy your code.
+# Add steps that build, run tests, deploy, and more:
+# https://aka.ms/yaml
+
+variables:
+  # The $(FlutterToolPath) variable is generated by the FlutterInstall task 
+  # See code in the Flutter Build task https://github.com/aloisdeniel/vsts-flutter-tasks/blob/master/tasks/build/index.ts
+  flutterExecPath: $(FlutterToolPath)/flutter.bat
+
+trigger:
+  batch: 'true'
+  branches:
+    include:
+      - develop
+      - release
+      - hotfix
+      - master
+  paths:
+    include:
+      - device_calendar/*
+
+stages:
+  # ----------- CI -----------
+  - stage: Build
+    jobs: 
+     - job: BuildAndroidAndIos
+       pool:
+        vmImage: 'macOS-latest'
+       steps:
+        - task: FlutterInstall@0
+          displayName: 'Flutter install'
+          inputs:
+            channel: 'stable'
+            version: 'latest'
+
+        - task: FlutterBuild@0
+          displayName: 'Flutter build - Android'
+          inputs:
+            target: 'aab'
+            projectDirectory: 'device_calendar/example'
+
+        - task: FlutterBuild@0
+          displayName: 'Flutter build - iOS'
+          inputs:
+            target: 'ios'
+            projectDirectory: 'device_calendar/example'
+            iosCodesign: false
+            
+  # -----------CD Pre-release -----------
+  - stage: DevelopmentRelease
+    dependsOn: Build
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/release'), eq(variables['Build.SourceBranch'], 'refs/hotfix'))
+    jobs:
+     - job: ReleaseAsDevelopmentVersion
+       pool:
+        vmImage: 'windows-latest'
+       steps:
+        - task: FlutterInstall@0
+          displayName: 'Flutter install'
+          inputs:
+            channel: 'stable'
+            version: 'latest'
+
+        - task: PowerShell@2
+          condition: eq(variables['Build.SourceBranch'], 'refs/release')
+          displayName: 'Pre-release versioning - pubspec.yaml'
+          inputs:
+            targetType: 'inline'
+            script: |
+              [string] $version = select-string -path "$(Build.SourcesDirectory)/device_calendar/pubspec.yaml" -pattern "(?<=version: ).*" | %{ $_.Matches[0].Value }
+              Write-Host "##vso[task.setvariable variable=currentVersion]$version"
+
+              [string] $versionRegex = '(?<=version: ).*'
+              [string] $pubspecContent = Get-Content -Path "$(Build.SourcesDirectory)/device_calendar/pubspec.yaml" -Raw
+
+              Write-Host " (i) Current version: $version"
+              Write-Host " (i) Original pubspec.yaml content: $pubspecContent"
+
+              [bool] $hasMatches = $pubspecContent -match $versionRegex
```

Comment:
Thanks for that. Removed the `-match-` script

Reply:
I would probably leave that code in (move it around though), as it might be helpful when reading logs to see if there were any matches or not. If you remove it then in case there are no matches you will get an exception message ( this will throw an exception `$_.Matches[0].Value`) which might not necessarily be very descriptive and harder to debug.